compiler/sem/semtypes.nim

```diff
@@ -986,7 +984,7 @@ proc skipGenericInvocation(t: PType): PType {.inline.} =
 
 proc addInheritedFields(c: PContext, check: var IntSet, pos: var int,
                         obj: PType) =
-  assert obj.kind == tyObject
+  assert obj.kind == tyObject, $obj.kind
```

Comment:
```suggestion
  assert obj.kind == tyObject
```

Reply:
wouldn't it be easier to debug by leaving the kind in there?

Reply:
It would, although I personally think that assertions with `$x` are visually a bit noisy, hence my suggestion. It's nothing critical, however, and I'm fine with it staying as it is.

It's nothing for this PR, but since asserting that the discriminator of some object has a certain value is a common pattern in the compiler, I think that adding a new idiom `assertKind` (which automatically renders the value on failure) would make sense.